js/express/src/CDK.ts

```diff
@@ -0,0 +1,141 @@
+  private environment: string;
+  private defaultLabels?: Record<string, string>;
+  private requestsCounterConfig: CounterConfiguration<string>;
+  private requestDurationHistogramConfig: HistogramConfiguration<string>;
+
+  private metricsServer: Express;
+  private requestsCounter?: Counter;
+  private requestsDurationHistogram?: Histogram;
+
+  constructor(options?: CDKOptions) {
+    // Initializing all the options
+    this.path = options?.path ?? "/metrics";
+    this.metricsServerPort = options?.metricsServerPort ?? 9097;
+    this.environment = options?.environment ?? "production";
+    this.defaultLabels = options?.defaultLabels;
+    this.requestsCounterConfig = options?.requestsCounterConfig ?? {
+      name: "http_requests_total",
+      help: "Total number of requests",
+      labelNames: ["path", "method", "status"],
+    };
+    this.requestDurationHistogramConfig =
+      options?.requestDurationHistogramConfig || {
+        name: "http_requests_duration_milliseconds",
+        help: "Duration of HTTP requests in milliseconds",
+        labelNames: ["path", "method", "status"],
+        buckets: promClient.exponentialBuckets(0.25, 1.5, 31),
+      };
+
+    // Create the metrics server using express
+    this.metricsServer = express();
+
+    this.initiateMetricsRoute();
+    this.initiatePromClient();
+  }
+
+  private initiatePromClient = () => {
+    promClient.register.setDefaultLabels({
+      environment: this.environment,
+      program: packageJson.name,
+      version: packageJson.version,
+      host: os.hostname(),
+      ip: getHostIpAddress(),
+      ...this.defaultLabels,
+    });
+
+    promClient.collectDefaultMetrics({
+      gcDurationBuckets: this.requestDurationHistogramConfig.buckets,
+    });
+
+    // Initiate the Counter for the requests
+    this.requestsCounter = new promClient.Counter(this.requestsCounterConfig);
+    // Initiate the Duration Histogram for the requests
+    this.requestsDurationHistogram = new promClient.Histogram(
+      this.requestDurationHistogramConfig
+    );
+  };
+
+  private initiateMetricsRoute = () => {
+    // Adding merics route handler
+    this.metricsServer.get(this.path, async (req, res) => {
+      // Adding Content-Type header
+      res.set("Content-Type", promClient.register.contentType);
+      return res.end(await promClient.register.metrics());
+    });
+
+    // Listening metrics server
+    this.metricsServer.listen(this.metricsServerPort, () => {
+      console.log(`Metrics server started at port ${this.metricsServerPort}`);
+    });
+  };
+
+  // Middleware Function, which is essentially the response-time middleware with a callback that captures the
+  // metrics
+  public REDMiddleware = ResponseTime(
+    (
+      req: IncomingMessage,
+      res: ServerResponse<IncomingMessage>,
+      time: number
+    ) => {
+      if (this.path !== req.url) {
+        const parsedPathname = getParsedPathname(req.url ?? "/", undefined);
```

Comment:
@chtushar can we use req.path instead of req.url. This just gives the path minus query params or anything extra.